packages/neuron-wallet/src/services/wallets.ts

```diff
@@ -371,18 +371,55 @@ export default class WalletService {
     const paths = addressInfos.map(info => info.path)
     const pathAndPrivateKeys = this.getPrivateKeys(wallet, paths, password)
 
-    const witnesses: string[] = inputs!.map((input: Input) => {
+    const witnessesWithLockHashes = inputs!.map((input: Input) => {
       const blake160: string = input.lock!.args!
+      const witnessArgs: WitnessArgs = {
+        lock: undefined,
+        inputType: undefined,
+        outputType: undefined
+      }
+      return {
+        // TODO: fill in required DAO's type witness here
+        witnessArgs,
+        lockHash: input.lockHash!,
+        witness: '',
+        blake160,
+      }
+    })
+
+    const lockHashes = new Set(witnessesWithLockHashes.map(w => w.lockHash))
+
+    for (const lockHash of lockHashes) {
+      const firstIndex = witnessesWithLockHashes.findIndex(w => w.lockHash === lockHash)
+      const witnessesArgsWithBlake160 = witnessesWithLockHashes
```

Comment:
Now there are `witness`, `witnesses`, `witnessArgs`, `witnessesArgs`, which is bewildering.

Reply:
`witnessesWithLockHashes` becomes a bad name now that there're more members.. If I understand how it was named previously, it should be called `witnessesWithLockHashesAndWitnessArgsAndBlake160` now? :P

Reply:
Or just witnessGenerationData for short